Task2-BasicWebApiCRUD/Controllers/ToDoController.cs

```diff
@@ -0,0 +1,91 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Todo.Application.Interfaces;
+using Todo.Domain.Models;
+
+namespace Task2_BasicWebApiCRUD.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class ToDoController : ControllerBase
+    {
+        private readonly ITodoService _todoService;
+        public ToDoController(ITodoService todoService)
+        {
+            _todoService = todoService;
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> AddItem(ToDoModel model)
+        {
+            try
+            {
+                var addedItem = await _todoService.AddItem(model);
+                return Ok(addedItem);
```

Comment:
Implement appropriate error handling for cases where the requested resource is not found or there is a validation error.
Provide appropriate HTTP status codes and responses for each endpoint.

Where are we handling this?

Reply:
<img width="455" alt="Screenshot 2023-10-13 161748" src="https://github.com/mjunaid-cnc/Talent-Pool-Training/assets/147056363/977f2a9a-ef7e-4022-b6a7-d0209a2b53b0">

Please see screenshot for error handling for the case where the resource is not found.